### PR TITLE
Cherry pick 1518: construct synthetic contract result for HTS

### PR DESCRIPTION
**Pull request by AlfredoG87**
**Description**:
Cherry pick 1518: construct synthetic contract result for HTS

**Related issue(s)**:
#1527

### CODE REVIEW

**Review by ebadiere — DISMISSED** [review]

File: packages/relay/src/lib/eth.ts [hunk 1]
```diff
@@ -1646,6 +1678,7 @@ export class EthImpl implements Eth {
     }
 
     await this.batchGetAndPopulateContractResults(contractResults, showDetails, transactionObjects, transactionHashes, requestIdPrefix);
+    await this.filterAndPopulateSyntheticContractResults(showDetails, logs, transactionObjects, transactionHashes, requestIdPrefix);
```
Comment (Nana-EC) [inline packages/relay/src/lib/eth.ts:1681]: Let's feature gate this if we're cherry picking.
Default to false though

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Feature gate with default off

File: packages/relay/src/lib/eth.ts (hunk 1 above)
Comment (AlfredoG87) [reply packages/relay/src/lib/eth.ts:1681]: should the feature gate be only on the `0.28` branch or should I include it in main too?

Also, do you think we should cherry pick it for `0.28` or is it better to wait for `0.29` to include it?

Comment (AlfredoG87) [reply packages/relay/src/lib/eth.ts]: ✅  Done

**Review by Nana-EC — APPROVED** [review]
LG

**Pull request merged**